lib/bus.js

```diff
@@ -160,6 +161,38 @@ Bus.prototype.validate = validateAndCheckGateway;
 Bus.prototype.toString = Object
   .getOwnPropertyDescriptor(Device.prototype, 'address').get;
 
+Bus.prototype.scan = function scan(timeout, callback) {
+
+  // Establish an empty device list
+  const devices = [];
+
+  this.on("command", function(command) {
+
+    var deviceId = `${command.sender.subnet}.${command.sender.id}`
+    // Check to see if this device is registered in the devices list
+    knownDevices = Devicemapping[command.sender.type]
+    foundDevice = {
+      info: knownDevices,
+      hdlType: command.sender.type,
+      address: deviceId,
+    }
+    // Add to device list
+    devices.push(foundDevice);
+
+  });
+
+  this.send("255.255", 0x000e, function(err) {
+    if (err) {
+      callback(err);
+    }
+  });
+
+  setTimeout(() => {
+    callback(null, devices);
```

Comment:
Don't forget to cleanup `command` event listener function before invoking callback. Currently you have a memory leak in this code.

Also consider to add a separate test for this behaviour.